https://bugs.gentoo.org/show_bug.cgi?id=417401

The Prefix toolchain on OSX doesn't support -arch, even though using
gcc-apple.  This is not really a problem (FSF GCC doesn't understand
this flag either), because we never build FAT objects.  It is however a
problem when people forcibly add -arch to CFLAGS and LDFLAGS.  Leave it
to the toolchain such that we can also build 64-bits.  (Pro!)

Request for removing this when unnecessary:
http://code.google.com/p/gyp/issues/detail?id=260

--- build/gyp/pylib/gyp/xcode_emulation.py
+++ build/gyp/pylib/gyp/xcode_emulation.py
@@ -317,7 +317,6 @@
       # TODO: Supporting fat binaries will be annoying.
       self._WarnUnimplemented('ARCHS')
       archs = ['i386']
-    cflags.append('-arch ' + archs[0])
 
     if archs[0] in ('i386', 'x86_64'):
       if self._Test('GCC_ENABLE_SSE3_EXTENSIONS', 'YES', default='NO'):
@@ -435,7 +434,6 @@
       # TODO: Supporting fat binaries will be annoying.
       self._WarnUnimplemented('ARCHS')
       archs = ['i386']
-    ldflags.append('-arch ' + archs[0])
 
     # Xcode adds the product directory by default.
     ldflags.append('-L' + product_dir)
